packages/babel-plugin-transform-optional-chaining/src/transform.ts

```diff
@@ -209,9 +212,10 @@ export function transformOptionalChain(
     !ifNullishBoolean && t.isUnaryExpression(ifNullish, { operator: "void" });
 
   const isEvaluationValueIgnored =
-    (t.isExpressionStatement(replacementPath.parent) ||
-      t.isSequenceExpression(replacementPath.parent)) &&
-    !replacementPath.isCompletionRecord();
+    (t.isExpressionStatement(replacementPath.parent) &&
+      !replacementPath.isCompletionRecord()) ||
+    (t.isSequenceExpression(replacementPath.parent) &&
+      last(replacementPath.parent.expressions) !== replacementPath.node);
```

Comment:
It might make sense to have a `path.isValueObservable()` helper.

Reply:
It certainly reminds me of the messy [`path._getCompletionRecords`](https://github.com/babel/babel/blob/c71a2177639b0782b594a44afd685d4631884e6d/packages/babel-traverse/src/path/family.ts#L226). Thankfully the switch statement is not an expression.

Reply:
Similar usage:
https://github.com/babel/babel/blob/c71a2177639b0782b594a44afd685d4631884e6d/packages/babel-plugin-proposal-destructuring-private/src/index.ts#L162-L164